### DISCUSSION TITLE
Getting all Items on a page of a Channel, and how to exclude Shorts

### DISCUSSION THREAD

**Discussion by BourgeoisDirk** [Tyrrrz/YoutubeExplode#685] [Q&A] [answered]
Hey, I'm trying to figure out how I can for instance get all the video's on a Channel's Page.
For example, [https://www.youtube.com/@PsyChamber](https://www.youtube.com/@PsyChamber) has a page [Videos](https://www.youtube.com/@PsyChamber/videos), and a page [Shorts](https://www.youtube.com/@PsyChamber/shorts), and a page [Playlists](https://www.youtube.com/@PsyChamber/playlists)
How would I go about downloading all items on one of those Pages ? (Playlists is not a problem)
Right now I just get the channel by their handle, and download all videos from said channel using the channelid, but this Includes Shorts.
- How would i go about to for instance only download all videos on the [Videos](https://www.youtube.com/@PsyChamber/videos) page (So excluding the Shorts) ?
- How would I go about only downloading the Shorts ?
Thanks

**Comment by Tyrrrz** [comment]
YouTube doesn't have a strict distinction between normal videos and shorts. It treats any video as a short if it's 1 minute or shorter, and uses vertical aspect ratio. So you can use these criteria to filter out videos.

**Comment by BourgeoisDirk** [comment, marked as answer]
Thanks @Tyrrrz, so then this should filter out the things i don't want ^^

var channel = await Client.Channels.GetByHandleAsync(String.Format("https://youtube.com/{0}", handle), CancellationToken);
var videos = await Client.Channels.GetUploadsAsync(channel.Id, CancellationToken);
videos = videos.Where(x => !x.Duration.Equals(null)).ToList();  // Remove Live Streams
videos = videos.Where(x => x.Duration.Value.Ticks > TimeSpan.TicksPerMinute).ToList();  // Remove Shorts

Much appreciate the pointer, have a good one !

**Reply by Tyrrrz** [reply]
That works but I would write it like this:

```csharp
var videos = (await Client.Channels.GetUploadsAsync(channel.Id, CancellationToken))
    .Where(x => x.Duration is not null && x.Duration.TotalMinutes > 1)
    .ToArray();
```

**Reply by BourgeoisDirk** [reply]
Agreed, and applied. Thanks !
